From 1c34f1ef9759cbad61546947f41e8f055cae20bb Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 20 Feb 2024 13:01:37 +1100
Subject: [PATCH 2/2] Revert "GH-37803: [CI][Dev][Python] Release and merge
 script errors (#37819)"

This reverts commit 79e49dbfb71efc70555417ba19cb612eb50924e8.
---
 ci/conda_env_archery.txt                    | 2 +-
 ci/conda_env_crossbow.txt                   | 2 +-
 ci/conda_env_python.txt                     | 2 +-
 dev/archery/setup.py                        | 2 +-
 dev/tasks/conda-recipes/arrow-cpp/meta.yaml | 4 ++--
 python/pyproject.toml                       | 2 +-
 python/requirements-build.txt               | 2 +-
 python/requirements-wheel-build.txt         | 2 +-
 python/setup.py                             | 2 +-
 9 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/ci/conda_env_archery.txt b/ci/conda_env_archery.txt
index 40875e0a5..ace7a42ac 100644
--- a/ci/conda_env_archery.txt
+++ b/ci/conda_env_archery.txt
@@ -25,7 +25,7 @@ jira
 pygit2
 pygithub
 ruamel.yaml
-setuptools_scm<8.0.0
+setuptools_scm
 toolz
 
 # benchmark
diff --git a/ci/conda_env_crossbow.txt b/ci/conda_env_crossbow.txt
index 59b799720..347294650 100644
--- a/ci/conda_env_crossbow.txt
+++ b/ci/conda_env_crossbow.txt
@@ -21,5 +21,5 @@ jinja2
 jira
 pygit2
 ruamel.yaml
-setuptools_scm<8.0.0
+setuptools_scm
 toolz
diff --git a/ci/conda_env_python.txt b/ci/conda_env_python.txt
index 972034421..4018ed200 100644
--- a/ci/conda_env_python.txt
+++ b/ci/conda_env_python.txt
@@ -28,4 +28,4 @@ pytest-faulthandler
 pytest-lazy-fixture
 s3fs>=2023.10.0
 setuptools
-setuptools_scm<8.0.0
+setuptools_scm
diff --git a/dev/archery/setup.py b/dev/archery/setup.py
index 2ecc72e04..02a8b3429 100755
--- a/dev/archery/setup.py
+++ b/dev/archery/setup.py
@@ -30,7 +30,7 @@ jinja_req = 'jinja2>=2.11'
 extras = {
     'benchmark': ['pandas'],
     'crossbow': ['github3.py', jinja_req, 'pygit2>=1.6.0', 'requests',
-                 'ruamel.yaml', 'setuptools_scm<8.0.0'],
+                 'ruamel.yaml', 'setuptools_scm'],
     'crossbow-upload': ['github3.py', jinja_req, 'ruamel.yaml',
                         'setuptools_scm'],
     'docker': ['ruamel.yaml', 'python-dotenv'],
diff --git a/dev/tasks/conda-recipes/arrow-cpp/meta.yaml b/dev/tasks/conda-recipes/arrow-cpp/meta.yaml
index b8ffbfdb7..8ceada1af 100644
--- a/dev/tasks/conda-recipes/arrow-cpp/meta.yaml
+++ b/dev/tasks/conda-recipes/arrow-cpp/meta.yaml
@@ -240,7 +240,7 @@ outputs:
         - numpy
         - python
         - setuptools
-        - setuptools_scm <8.0.0
+        - setuptools_scm
       run:
         - {{ pin_subpackage('libarrow', exact=True) }}
         - {{ pin_compatible('numpy') }}
@@ -322,7 +322,7 @@ outputs:
         - numpy
         - python
         - setuptools
-        - setuptools_scm <8.0.0
+        - setuptools_scm
       run:
         - {{ pin_subpackage('pyarrow', exact=True) }}
         - python
diff --git a/python/pyproject.toml b/python/pyproject.toml
index 437de105a..fe8c938a9 100644
--- a/python/pyproject.toml
+++ b/python/pyproject.toml
@@ -19,7 +19,7 @@
 requires = [
     "cython >= 0.29.31",
     "oldest-supported-numpy>=0.14",
-    "setuptools_scm < 8.0.0",
+    "setuptools_scm",
     "setuptools >= 40.1.0",
     "wheel"
 ]
diff --git a/python/requirements-build.txt b/python/requirements-build.txt
index 56e9d479e..507e90813 100644
--- a/python/requirements-build.txt
+++ b/python/requirements-build.txt
@@ -1,4 +1,4 @@
 cython>=0.29.31
 oldest-supported-numpy>=0.14
-setuptools_scm<8.0.0
+setuptools_scm
 setuptools>=38.6.0
diff --git a/python/requirements-wheel-build.txt b/python/requirements-wheel-build.txt
index f42ee4a01..6043d2ffb 100644
--- a/python/requirements-wheel-build.txt
+++ b/python/requirements-wheel-build.txt
@@ -1,5 +1,5 @@
 cython>=0.29.31
 oldest-supported-numpy>=0.14
-setuptools_scm<8.0.0
+setuptools_scm
 setuptools>=58
 wheel
diff --git a/python/setup.py b/python/setup.py
index eb9f72ac3..49e2dea15 100755
--- a/python/setup.py
+++ b/python/setup.py
@@ -492,7 +492,7 @@ setup(
                                  'pyarrow/_generated_version.py'),
         'version_scheme': guess_next_dev_version
     },
-    setup_requires=['setuptools_scm < 8.0.0', 'cython >= 0.29.31'] + setup_requires,
+    setup_requires=['setuptools_scm', 'cython >= 0.29.31'] + setup_requires,
     install_requires=install_requires,
     tests_require=['pytest', 'pandas', 'hypothesis'],
     python_requires='>=3.8',
